packages/yew/src/callback.rs

```diff
@@ -46,15 +50,31 @@ macro_rules! generate_callback_impls {
             }
         }
 
-        impl<IN> $callback<IN> {
+        impl<IN: 'static> $callback<IN> {
             /// Creates a "no-op" callback which can be used when it is not suitable to use an
             /// `Option<$callback>`.
             pub fn noop() -> Self {
-                Self::from(|_: $in_ty| ())
+                thread_local! {
+                    static NOOP: RefCell<HashMap<any::TypeId, Box<dyn any::Any>>> = RefCell::new(Default::default());
+                }
+
+                Self {
+                    cb: NOOP.with(|noop| {
+                        noop.borrow_mut()
+                            .entry(any::TypeId::of::<$in_ty>())
+                            .or_insert_with(|| {
+                                let func: Rc<dyn Fn($in_ty)> = Rc::new(|_: $in_ty| ());
+                                Box::new(func) as Box<dyn any::Any>
+                            })
+                            .downcast_ref::<Rc<dyn Fn($in_ty)>>()
+                            .unwrap()
+                            .clone()
+                    }),
+                }
```

Comment:
Is there really no other solution other than unwrapping and modifying a `HashMap` in a `RefCell`? that seems to me like too much work for a `Default::default()` value

Reply:
Since the function is doing nothing I think it might be possible to do something with unsafe that would be much faster? But I'm really no expert so I won't write that lol

One alternative would be to wrap the Rc into an Option like this:

```
pub struct Callback<IN, OUT = ()> {
    /// A callback which can be called multiple times
    pub(crate) cb: Option<Rc<dyn Fn(IN) -> OUT>>,
}
```

(Or using an enum instead of struct)

Reply:
But I'm afraid using Option will have a cost at runtime because every call will need to check the option.